tools/dep_updaters/update-libuv.sh

```diff
@@ -2,16 +2,29 @@
 set -e
 # Shell script to update libuv in the source tree to a specific version
 
-BASE_DIR=$(cd "$(dirname "$0")/../.." && pwd)
-DEPS_DIR="$BASE_DIR/deps"
-LIBUV_VERSION=$1
-
 if [ "$#" -le 0 ]; then
   echo "Error: please provide an libuv version to update to"
   echo "	e.g. $0 1.44.2"
   exit 1
 fi
 
+BASE_DIR=$(cd "$(dirname "$0")/../.." && pwd)
+DEPS_DIR="$BASE_DIR/deps"
+
+NEW_VERSION=$1
+VERSION_H="$DEPS_DIR/uv/include/uv/version.h"
+CURRENT_MAJOR_VERSION=$(grep "#define UV_VERSION_MAJOR" "$VERSION_H" | sed -n "s/^.*MAJOR \(.*\)/\1/p")
+CURRENT_MINOR_VERSION=$(grep "#define UV_VERSION_MINOR" "$VERSION_H" | sed -n "s/^.*MINOR \(.*\)/\1/p")
+CURRENT_PATCH_VERSION=$(grep "#define UV_VERSION_PATCH" "$VERSION_H" | sed -n "s/^.*PATCH \(.*\)/\1/p")
+CURRENT_SUFFIX_VERSION=$(grep "#define UV_VERSION_SUFFIX" "$VERSION_H" | sed -n "s/^.*SUFFIX \"\(.*\)\"/\1/p")
+SUFFIX_STRING=$([ -z "$CURRENT_SUFFIX_VERSION" ] && echo "" || echo "-$CURRENT_SUFFIX_VERSION")
+CURRENT_VERSION="$CURRENT_MAJOR_VERSION.$CURRENT_MINOR_VERSION.$CURRENT_PATCH_VERSION$SUFFIX_STRING"
```

Comment:
I realize it's existing code that's been moved around but it might be simpler to ignore UV_VERSION_SUFFIX and instead check that UV_VERSION_IS_RELEASE is 1. The former is always an empty string when the latter is 1.

Reply:
I try [this](https://github.com/nodejs/node/pull/46488/commits/16944863ab39491c8f641f22394e4099b4db1c06); let me know if it matches the requirements.

Reply:
Hey @bnoordhuis 👋🏼 

Could you please let me know if it looks okay on your side?